ST_mortality.c

```diff
@@ -204,67 +204,135 @@ void mort_Main( Bool *killed) {
-    GroupType *g;
+    GroupType *g = NULL;
     SppIndex sp;
     IntU j;
-
-    ForEachGroup(rg) {
-        if (Globals.currYear < RGroup[rg]->startyr) {
-            /* Can't kill plants until the year each RGroup[rg] is turned on */
-            continue;
+    RealF fire_possibility, random_number, biomass_cheatgrass;
+    char *cheatgrass_name = "brte";
+    int i = 0;
+    
+    /* Check species index number from the beginning to all the species in
+     *  species.in , if the species name == checkname then get the biomass and stop the loop*/
+    for (i = 0; i < Globals.sppCount; i++) { /* if species name = checkname = brte then get the biomass of brte(cheatgrass)*/
+        if (strcmp(cheatgrass_name, Species[i]->name) == 0) {
+            biomass_cheatgrass = Species_GetBiomass(i); /* calculate biomass of cheatgrass*/
+            //  printf("Cheatgrass: %s\n",Species[i]->name);
+            g = RGroup[Species[i]->res_grp];
+            break;
         }
+    }
+    
+    /* Set a random number outside of the loop to make sure the kill probability for each functional group is the same */
+    random_number = RandUni(&mortality_rng);
+    //printf("[Cheatgrass: %f\n",biomass_cheatgrass);
 
-        g = RGroup[rg];
-
-        if ((Globals.currYear >= g->killfreq_startyr) && (g->killfreq > 0.)) {
-            if(g->killfreq < 1.0){
-                if (RandUni(&mortality_rng) <= g->killfreq) {
-                    g->killyr = Globals.currYear;
-                }
-            } else if (((Globals.currYear - g->killfreq_startyr) % (IntU) g->killfreq) == 0) {
-                g->killyr = Globals.currYear;
-            }
+    // in this for loop "g" refers to the RGroup of cheatgrass. RGroup[rg] refers
+    // to the current iteration's RGroup.
+    ForEachGroup(rg) {
+      if (Globals.currYear < RGroup[rg]->startyr) {
+        continue;
+      }
 
+      RGroup[rg]->prescribedfire = 0;
+      RGroup[rg]->wildfire = 0;
+
+      // If these conditions are true we want to simulate wildfire
+      if(g != NULL && g->killfreq == 0){
+        /* ------------------------- WILDFIRE SIMULATION ------------------------- */
+        // Calculate fire_possibility
+        if (g->ignition == 0) { 
+          /* If ignition == 0, no wildfire occurs */
+          fire_possibility = 0;
+        } else if (biomass_cheatgrass < g->ignition) {
+          /* If cheatgrass biomass is less than the biomass required for wildfire ignition, wildfire probability is very low*/
+          fire_possibility = .01;
+        } else { 
+          /* Otherwise a wildfire probability is calculated, which increases with cheatgrass biomass*/
+          fire_possibility = g->cheatgrass_coefficient + g->wild_fire_slope * biomass_cheatgrass;
+
+          // Cap fire_possibility at 1. This isn't needed from an algorithmic perspective,
+          // but a value greater than 1 does not make sense as a percentage.
+          if(fire_possibility > 1){
+            fire_possibility = 1;
```

Comment:
1 and 1.0 are are treated the same. Because fire_possibility is a float, 1 will be treated as a floating point value. There are a lot of places in the code that use 1.0 instead of 1 but in situations like this it isn't necessary.

Reply:
@kpalmqui For example, one situation where it would be necessary is arithmetic:
```
1 / 5 == 0 
```
because it uses integer division but
```
1.0 / 5.0 == .2
```
because it uses floating point division.

In logic this isn't necessary because the compiler decides what we mean when we say `1`. Since fire_possibility is a float the compiler will turn `1` into a float.